### PR TITLE
[javascript] CPD: Added support for decorator notation

### PR DESCRIPTION
**Pull request by wener-tiobe**
## Describe the PR

This is to resolve the following error on a javascript file:

[main] ERROR net.sourceforge.pmd.cli - Error while tokenizing: Lexical error in file '.\TableList.js' at
 line 67, column 1: "@" (64), after : "" (in lexical state DEFAULT)

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [x] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by wener-tiobe** [comment]
I'll add a unit test too.

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No regression tested rules have been changed.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-javascript/etc/grammar/Ecmascript5.jj [hunk 1]
```diff
@@ -281,7 +281,8 @@ TOKEN :
     |
         [
           "$",
-          "_"
+          "_",
+          "@"
```
Comment (adangel) [inline pmd-javascript/etc/grammar/Ecmascript5.jj:285]: I wouldn't add `@` as part of an identifier. According to the standard, it is separate - see https://arai-a.github.io/ecma262-compare/snapshot.html?pr=2417#sec-decorators
When keeping it separately, we could potentially identify duplicated identifier names.

File: pmd-javascript/src/test/resources/net/sourceforge/pmd/lang/ecmascript/cpd/testdata/decorator.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+    [Image] or [Truncated image[            Bcol      Ecol
+L2
+    [@annotation]                           1         12
```
Comment (adangel) [inline pmd-javascript/src/test/resources/net/sourceforge/pmd/lang/ecmascript/cpd/testdata/decorator.txt:3]: That should actually be 2 tokens: @ and Identifier

**Review by adangel — APPROVED** [review]
Thanks!

I'll change the grammar myself before merging.

In case, someone wonders: Decorators are an experimental feature of ecmascript not yet part of the standard, see https://github.com/tc39/proposal-decorators and https://github.com/tc39/ecma262/pull/2417

See https://arai-a.github.io/ecma262-compare/snapshot.html?pr=2417#sec-decorators for the proposed standard document.

**Pull request merged**